flink-connectors/flink-connector-elasticsearch-base/src/main/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorFactory.java

```diff
@@ -88,6 +89,21 @@ private static IndexGenerator createRuntimeIndexGenerator(
                 index.substring(indexPrefix.length() + dynamicIndexPatternStr.length());
 
         final boolean isDynamicIndexWithFormat = indexHelper.checkIsDynamicIndexWithFormat(index);
+        final boolean isDynamicIndexWithProcTimeFormat =
+                indexHelper.checkIsDynamicIndexWithProcTimeFormat(index);
+
+        if (isDynamicIndexWithProcTimeFormat) {
+            final String dateTimeFormat =
+                    indexHelper.extractDateFormat(
+                            index, LogicalTypeRoot.TIMESTAMP_WITHOUT_TIME_ZONE);
+            return new AbstractTimeIndexGenerator(index, dateTimeFormat) {
+                @Override
+                public String generate(RowData row) {
+                    return indexPrefix.concat(LocalDateTime.now().format(dateTimeFormatter));
```

Comment:
Here, we are using `now()` and not proctime, is that what we want?

Reply:
The return type of now() should be  TIMESTAMP_WITH_LTZ,  The LocalDate.now() will use jvm timezone.  To avoid misunderstanding， We should keep the same behavior.    But I'm not sure we can get the session timezone from connector?

Reply:
I found we can get timezone id from  table config from context.getConfiguration(), What do you think about this?  And I updated the code.

Reply:
Sorry, I forgot to reply here, but yep, what you did with the timezone from the table config is good!